overlord/assertstate/assertstate_test.go

```diff
@@ -2077,3 +2136,205 @@ func (s *assertMgrSuite) TestStore(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(store.Store(), Equals, "foo")
 }
+
+// validation-sets related tests
+
+func (s *assertMgrSuite) TestRefreshValidationSetAssertionsNop(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.setModel(sysdb.GenericClassicModel())
+
+	err := assertstate.RefreshValidationSetAssertions(s.state, 0)
+	c.Assert(err, IsNil)
+}
+
+func (s *assertMgrSuite) TestRefreshValidationSetAssertionsStoreError(c *C) {
+	s.fakeStore.(*fakeStore).snapActionErr = &store.UnexpectedHTTPStatusError{StatusCode: 400}
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.setModel(sysdb.GenericClassicModel())
+
+	tr := assertstate.ValidationSetTracking{
+		AccountID: "foo",
+		Name:      "bar",
+		Mode:      assertstate.Monitor,
+		Current:   1,
+	}
+	assertstate.UpdateValidationSet(s.state, &tr)
+
+	err := assertstate.RefreshValidationSetAssertions(s.state, 0)
+	c.Assert(err, ErrorMatches, `unsuccessful bulk assertion refresh, fallback: cannot : got unexpected HTTP status code 400.*`)
+}
+
+func (s *assertMgrSuite) TestRefreshValidationSetAssertions(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// have a model and the store assertion available
+	storeAs := s.setupModelAndStore(c)
+	err := s.storeSigning.Add(storeAs)
+	c.Assert(err, IsNil)
+
+	// store key already present
+	err = assertstate.Add(s.state, s.storeSigning.StoreAccountKey(""))
+	c.Assert(err, IsNil)
+
+	err = assertstate.Add(s.state, s.dev1Acct)
+	c.Assert(err, IsNil)
+
+	snapDeclFoo := s.snapDecl(c, "foo", nil)
+	err = assertstate.Add(s.state, snapDeclFoo)
+	c.Assert(err, IsNil)
+
+	// previous state
+	/*vsetAs1 := s.validationSetAssert(c, "bar", "1", "1")
+	err = assertstate.Add(s.state, vsetAs1)
+	c.Assert(err, IsNil)*/
+
+	vsetAs2 := s.validationSetAssert(c, "bar", "1", "1")
+	err = s.storeSigning.Add(vsetAs2)
+	c.Assert(err, IsNil)
+
+	tr := assertstate.ValidationSetTracking{
+		AccountID: s.dev1Acct.AccountID(),
+		Name:      "bar",
+		Mode:      assertstate.Monitor,
+		Current:   1,
+	}
+	assertstate.UpdateValidationSet(s.state, &tr)
+
+	err = assertstate.RefreshValidationSetAssertions(s.state, 0)
+	c.Assert(err, IsNil)
+
+	a, err := assertstate.DB(s.state).Find(asserts.ValidationSetType, map[string]string{
+		"series":     "16",
+		"account-id": s.dev1Acct.AccountID(),
+		"name":       "bar",
+		"sequence":   "1",
+	})
+	c.Assert(err, IsNil)
+	c.Check(a.(*asserts.ValidationSet).Name(), Equals, "bar")
+
+	// XXX: why do we get account keys?
```

Comment:
because we always try to update prerequisites?  or am I misunderstanding the question